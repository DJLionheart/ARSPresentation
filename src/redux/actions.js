import fetch from '../utils/fetch'

export const setLuke = (data) => ({
    type: types.LUKE,
    payload: data
})

export const fetchLuke = () =>
    async (disaptch) => {
        const response = await disaptch(fetch.get('https://swapi.co/api/people/1'))
        console.log('hit', response)
        disaptch(setLuke(response))
    }

export const types = {
    LUKE: 'LUKE'
}