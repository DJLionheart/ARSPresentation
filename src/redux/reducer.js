import { Map } from 'immutable'

import paths from './paths'
import { types } from './actions'

export default (state = Map(), action) => {
    const { type, payload } = action

    switch (type) {
        case types.LUKE:
            return state.mergeIn(paths.luke, payload)
        default:
            return state
    }
}