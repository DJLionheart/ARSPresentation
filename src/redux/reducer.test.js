import reducer from './reducer'
import { types } from './actions'

describe('reducer', () => {
    test('reducer returns luke', () => {
        const action = {
            type: types.LUKE,
            payload: {
                name: 'Luke Skywalker',
                from: 'Swapi'
            }
        }

        expect(reducer(undefined, action)).toMatchSnapshot()
    })
})