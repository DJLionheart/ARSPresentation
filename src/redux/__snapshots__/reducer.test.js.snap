// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer reducer returns luke 1`] = `
Immutable.Map {
  "luke": Immutable.Map {
    "name": "Luke Skywalker",
    "from": "Swapi",
  },
}
`;
